zppy/templates/coupled_global.py

```diff
@@ -662,210 +1205,194 @@ def make_plot_pdfs(
+        f"These {rgn} regions plots could not be generated successfully: {invalid_plots}"
     )
 
 
-def run_by_region(parameters):
-    regions = parameters[14].split(",")
-    for rgn in regions:
-        if rgn.lower() in ["glb", "global"]:
-            rgn = "glb"
-        elif rgn.lower() in ["n", "north", "northern"]:
-            rgn = "n"
-        elif rgn.lower() in ["s", "south", "southern"]:
-            rgn = "s"
+def get_vars(requested_variables: RequestedVariables, component: str) -> List[Variable]:
+    vars: List[Variable]
+    if component == "original":
+        vars = requested_variables.vars_original
+    elif component == "atm":
+        vars = requested_variables.vars_atm
+    elif component == "ice":
+        vars = requested_variables.vars_ice
+    elif component == "lnd":
+        vars = requested_variables.vars_land
+    elif component == "ocn":
+        vars = requested_variables.vars_ocn
+    else:
+        raise ValueError(f"Invalid component={component}")
+    return vars
+
+
+def create_viewer(parameters: Parameters, vars: List[Variable], component: str) -> str:
+    viewer = OutputViewer(path=".")
+    viewer.add_page("Table", parameters.regions)
+    groups: List[VariableGroup] = get_variable_groups(vars)
+    for group in groups:
+        # Only groups that have at least one variable will be returned by `get_variable_groups`
+        # So, we know this group will be non-empty and should therefore be added to the viewer.
+        viewer.add_group(group.group_name)
+        for var in group.variables:
+            plot_name: str = var.variable_name
+            row_title: str
+            if var.long_name != "":
+                row_title = f"{plot_name}: {var.long_name}"
+            else:
+                row_title = plot_name
+            viewer.add_row(row_title)
+            for rgn in parameters.regions:
+                # v3.LR.historical_0051_glb_lnd_SOIL4C.png
+                # viewer/c-state/glb_lnd_soil4c.html
+                viewer.add_col(
+                    f"{parameters.figstr}_{rgn}_{component}_{plot_name}.png",
+                    is_file=True,
+                    title=f"{rgn}_{component}_{plot_name}",
+                )
+
+    url = viewer.generate_page()
+    viewer.generate_viewer()
+    # Copy the contents of `table` into the `viewer` directory
+    # (which initially only has `css` and `js` subdirectories)
+    # Because `viewer` already exists,
+    # `shutil.copytree` will not work.
+    distutils.dir_util.copy_tree("table", "viewer")
+    print(
+        os.getcwd()
+    )  # /lcrc/group/e3sm/ac.forsyth2/zppy_min_case_global_time_series_viewers_output/test-pr616-20241022v2/v3.LR.historical_0051/post/scripts/global_time_series_1985-1995_dir
+    # shutil.rmtree("table")
+    # new_url = f"viewer_{component}"
+    # # shutil.move("viewer", new_url)
+    # distutils.dir_util.copy_tree("viewer", new_url)
```

Comment:
Any time I try to move/copy the land viewer over to a different dir (e.g., `viewer_lnd`), the formatting goes bad. I'm assuming some formatting file it's referencing isn't also getting copied over.

What this means though is that I can't seem to properly make different viewer directories for different components.